### PR TITLE
FOCUS #153: CommitmentTerm spec changes and supporting content

### PR DESCRIPTION
**Pull request by letian-wang-92**

### CODE REVIEW

File: specification/dimensions/commitmentterm.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Commitment Term
+
+A Commitment Term refers to the predetermined length of time under which a commitment-based discount applies. This term is agreed upon contractually and can vary depending on the pricing models, payment options, and specific requirements of the user and provider. During this term, the user commits to use a certain amount of a specific type of usage, such as virtual machines or databases, in exchange for a discounted unit price. The Commitment Term is an essential part of a commitment-based discount, dictating the duration of the contractual agreement. The duration of the term can be diverse, ranging from months to multiple years.
+
+The CommitmentTerm column MUST be present in the billing data. This column MUST be of type String and MUST NOT contain null values. The CommitmentTerm column MUST conform to the following format: Numberical value followed by a space and a time unit. The time unit MUST be one of the following allowed values: hour, day, week, month, year. The numerical value MUST be an integer. The CommitmentTerm column MUST NOT contain any other values than the ones specified above.
```
Comment (mike-finopsorg) [inline specification/dimensions/commitmentterm.md:5]: `MUST NOT contain null values` What is present for lines not related to a commitment?

File: specification/dimensions/commitmentterm.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Commitment Term
```
[the remark this answers is not in the document]
Comment (letian-wang-92) [reply specification/dimensions/commitmentterm.md:1]: Agree that this is a better name.

File: specification/dimensions/commitmentterm.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (letian-wang-92) [reply specification/dimensions/commitmentterm.md]: The ISO 8601 standard is actually how Azure currently handle this value. I personally like more machine readable format, happy with either

File: specification/dimensions/commitmentterm.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rileyjenk) [reply specification/dimensions/commitmentterm.md:1]: I believe that this level of detail for commitments is unnecessary in the focus spec. Providers already provide a historical record of commitments even after they are terminated. Provided the right IDs in our data, practitioners could easily join in data from their commitments to provide this information without natively including it in the spec.

**Comment by udam-f2** [comment]
@letian-wang-92 @flanakin What are the next steps on this?

**Review by jasonkellyAP — APPROVED** [review]
LGTM

**Comment by flanakin** [comment]
We decided to remove this column from 1.0 because the scenario that it was prioritized for (commitment planning) is more likely to use another data source to get the term. We still believe this is a useful column, but would like to spend more time and see how it relates to other columns that describe the thing you're being charged for (since this is aligned with those fields).

**Pull request closed without merging**